### PR TITLE
Fix multiple input-related problems on SDL

### PR DESCRIPTION
**Pull request by ThomasMiz**
# Summary of the PR
This PR fixes input events not being registered correctly and the window hanging problem from issue #698.

### CODE REVIEW

**Comment by ThomasMiz** [comment]
I think on line 96 of sdlView.cs I should add a '?' after System.Action. Kinda odd how nullable interacts with events huh

File: src/Input/Silk.NET.Input.Sdl/SdlKeyboard.cs [hunk 1]
```diff
@@ -60,13 +60,14 @@ public unsafe void DoEvent(Event @event)
...
-                    while (chars[i] != default)
+                    if (KeyChar != null)
                     {
-                        KeyChar?.Invoke(this, chars[i++]);
+                        var chars = stackalloc char[32];
+                        Encoding.UTF8.GetChars(&@event.Text.Text[0], 32, chars, 32);
+                            
+                        // run the KeyChar event until we get a null terminator or run out of buffer
+                        for (int i = 0; i < 32 && chars[i] != '\0'; i++)
+                            KeyChar.Invoke(this, chars[i]);
```
Comment (Perksey) [inline src/Input/Silk.NET.Input.Sdl/SdlKeyboard.cs:70]: nit: use braces

```suggestion
                        for (int i = 0; i < 32 && chars[i] != '\0'; i++)
                        {
                            KeyChar.Invoke(this, chars[i]);
                        }
```

File: src/Input/Silk.NET.Input.Sdl/SdlKeyboard.cs [hunk 2]
```diff
@@ -60,13 +60,14 @@ public unsafe void DoEvent(Event @event)
                 }
                 case EventType.Textinput:
                 {
-                    var chars = stackalloc char[32];
-                    Encoding.UTF8.GetChars(&@event.Text.Text[0], 32, chars, 32);
-                    int i = 0;
-                    // run the KeyChar event until we get a null terminator
-                    while (chars[i] != default)
+                    if (KeyChar != null)
```
Comment (Perksey) [inline src/Input/Silk.NET.Input.Sdl/SdlKeyboard.cs:63]: nit: prefer identity comparisons

```suggestion
                    if (KeyChar is not null)
```

**Review by Perksey — APPROVED** [review]
Few nitpicks for you to look at before you hit the merge button, but I'm happy with this.

File: src/Windowing/Silk.NET.Windowing.Sdl/SdlView.cs [hunk 3]
```diff
@@ -93,6 +93,7 @@ protected override Vector2D<int> CoreSize
             }
         }
 
+        public event System.Action ProcessingEvents;
```
Comment (ThomasMiz) [inline src/Windowing/Silk.NET.Windowing.Sdl/SdlView.cs:96]:
```suggestion
        public event System.Action? ProcessingEvents;
```

**Pull request merged**